### ISSUE TITLE
[type] Unexpected serialization of instance with circular ref

### ISSUE THREAD

**Issue by lionelhorn** [deepkit/deepkit-framework#85]
Considering the following types, I can't get it to properly serialize.
The serialized object still contains typed instances.

```javascript

export class Item {
  @t.array(ModifierGroup)
  modifierGroups: ModifierGroup[];

  @t id: string;

  @t.type(() => Item).parentReference
  @t.group("exclude")
  parent: Item;
}

export class ModifierGroup {
  @t.array(Item)
  modifiers: Item[];

  @t id: string;
}

export function serializeWithDeepKit() {
  const i1 = new Item();
  i1.id = "burger";

  const s1 = new Item();
  s1.id = "ketchup";
  s1.parent = i1;

  const s2 = new Item();
  s2.id = "mayo";
  s2.parent = i1;

  const mg = new ModifierGroup();
  mg.modifiers = [s1, s2];
  mg.id = "sauces";

  i1.modifierGroups = [mg];

  const serialized = jsonSerializer.for(Item).serialize(i1, {groupsExclude: ["exclude"]});
  try {
    console.log(JSON.stringify(serialized));
  } catch (e) {
    console.log(e);
  }
}
```
![image](https://user-images.githubusercontent.com/50528276/104629157-676b4380-5699-11eb-9b82-51a1139e9620.png)

I can't stringify the resulting object because of the circular ref.
```code
TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'ModifierGroup'
    |     property 'modifiers' -> object with constructor 'Array'
    |     index 0 -> object with constructor 'Item'
    |     ...
    |     property 'modifierGroups' -> object with constructor 'Array'
    --- index 0 closes the circle
    at JSON.stringify (<anonymous>)
    ````

**Comment by lionelhorn** [comment]
With the following simpler example (and no circular ref), serialization removed the parent field.

```javascript
export function serializeWithDeepKitSimple() {
  const i1 = new Item();
  i1.id = "burger";

  const s1 = new Item();
  s1.id = "ketchup";
  s1.parent = i1;

  const serialized = jsonSerializer.for(Item).serialize(s1, {groupsExclude: ["exclude"]});
  try {
    console.log(JSON.stringify(serialized));
  } catch (e) {
    console.log(e);
  }
}
```

`serialized => {  "id": "ketchup"}`

**Comment by marcj** [comment]
Fixed in master. Circular references are automatically removed now.

**Comment by marcj** [comment]
released in v1.0.1-alpha.13
